pkg/virt-operator/webhooks/kubevirt-update-admitter.go

```diff
@@ -433,18 +433,18 @@ func validateInfraReplicas(replicas *uint8) []metav1.StatusCause {
 }
 
 func featureGatesChanged(currKVSpec, newKVSpec *v1.KubeVirtSpec) bool {
-	currDevConfig := currKVSpec.Configuration.DeveloperConfiguration
-	newDevConfig := newKVSpec.Configuration.DeveloperConfiguration
-
-	if (currDevConfig == nil && newDevConfig == nil) || (currDevConfig != nil && newDevConfig == nil) {
-		return false
+	var curKvLegacyFGs, newKvLegacyFGs []string
+	if currKVSpec.Configuration.DeveloperConfiguration != nil {
+		curKvLegacyFGs = currKVSpec.Configuration.DeveloperConfiguration.FeatureGates
 	}
-
-	if currDevConfig == nil && newDevConfig != nil {
-		return len(newDevConfig.FeatureGates) > 0
+	if newKVSpec.Configuration.DeveloperConfiguration != nil {
+		newKvLegacyFGs = newKVSpec.Configuration.DeveloperConfiguration.FeatureGates
 	}
 
-	return !equality.Semantic.DeepEqual(currDevConfig.FeatureGates, newDevConfig.FeatureGates)
+	curFGs := featuregate.GetEnabledFeatureGates(currKVSpec.Configuration.FeatureGates, curKvLegacyFGs)
+	newFGs := featuregate.GetEnabledFeatureGates(newKVSpec.Configuration.FeatureGates, newKvLegacyFGs)
+
+	return !equality.Semantic.DeepEqual(curFGs, newFGs)
```

Comment:
I don't like this.
`featureGatesChanged` function is used to warning a deprecation.
I think the warning message should be prompted everytime the user change a something related to FG.

Reply:
Maybe it's just a matter of taste :)
Let's leave it as is.
/lgtm
Thanks